lang_ruby/parsing/parse_ruby.ml

```diff
@@ -115,9 +117,13 @@ let parse file =
       (* -------------------------------------------------- *)
       let lst =
         (* GLR parsing can be very time consuming *)
-        Common.timeout_function 10 (fun () ->
-          Parser_ruby.main lexer lexbuf
-        )
+        match
+          Common.set_timeout ~name:"Parse_ruby.parse" 10.0 (fun () ->
```

Comment:
May this `set_timeout` conflict with the one we have in Semgrep via `Main.iter_files_and_get_matches_and_exn_to_errors` ?

Reply:
It will. What I do for JS is that there is a timeout parameter to the parsing function, so we can disable it in the caller
by passing a value of 0.